source/speech/speech.py

```diff
@@ -960,16 +965,15 @@ def speak(  # noqa: C901
 	@param symbolLevel: The symbol verbosity level; C{None} (default) to use the user's configuration.
 	@param priority: The speech priority.
 	"""
+	pre_filter_speechSequence.notify(sequence=speechSequence)
 	speechSequence = filter_speechSequence.apply(speechSequence)
 	logBadSequenceTypes(speechSequence)
 	# in case priority was explicitly passed in as None, set to default.
 	priority: Spri = Spri.NORMAL if priority is None else priority
 
 	if not speechSequence:  # Pointless - nothing to speak
 		return
-	import speechViewer
-	if speechViewer.isActive:
-		speechViewer.appendSpeechSequence(speechSequence)
+	post_filter_speechSequence.notify(sequence=speechSequence)

```

Comment:
Shouldn't this be after the speech mode checks (the if statements below it)?
If those trigger, nothing is spoken. Therefore nothing should appear in the viewer, nothing will have been spoken.

Reply:
I believe this has been discussed here before, but to recap this notification has to occur **before** checking what speech mode is active, to support the following use cases:
- As a sighted tester who generally works with speech disabled I want to know what would have been spoken
- As a Braille user I disable the speech, but want to see it on my Braille display in some cases (this, of course, will be possible only after #15898 is fixed)
- If this extension point would be useful for a remote access solutions it is also important the notification is send even in cases in which the person accepting help disabled the speech, i.e. since they're sighted and find the fact that their computer talks during the remote session annoying.

In general there should be no change in logic here.